frontend/src/pages/chat/chat.tsx

```diff
@@ -1,16 +1,23 @@
 import { BackButton } from '#libs/components/components.js';
+import { AppRoute } from '#libs/enums/enums.js';
 import { getValidClassNames } from '#libs/helpers/helpers.js';
-import { useCallback, useSidebarState } from '#libs/hooks/hooks.js';
+import {
+  useCallback,
+  useNavigate,
+  useSidebarState,
+} from '#libs/hooks/hooks.js';
 
 import { ChatLayout, ChatSidebar } from './components/components.js';
 import styles from './styles.module.scss';
 
 const Chat: React.FC = () => {
+  const navigate = useNavigate();
   const { isSidebarShown, setIsSidebarShown } = useSidebarState();
 
   const handleBackButtonPress = useCallback(() => {
+    navigate(AppRoute.CHATS);
```

Comment:
But we have appActions for such purpose

Reply:
unfortunately, it doesn't work properly like this. It doesn't add the query string after redirection with `appActions`

Reply:
But ithere is no query string in `navigate(AppRoute.CHATS);`

Reply:
Then how does it work?

Reply:
yep, but there is `setIsSidebarShown(true)` below

Reply:
I guess navigate in appActions rewrites the URL after applying the query string

Reply:
navigate in appActions has the same logic as yours navigate. Btw, I've checked locally and both solutions work equally

Reply:
hm, that's strange, cause I've also tested. Did you write like this?
```ts
  const handleBackButtonPress = useCallback(() => {
    dispatch(appActions.navigate(AppRoute.CHATS));
    setIsSidebarShown(true);
  }, [setIsSidebarShown, dispatch]);
```

Reply:
Ok, let's proceed with your solution, as we have not enough time to re-check it